modules/ducktests/tests/ignitetest/tests/discovery_test.py

```diff
@@ -149,18 +179,134 @@ def _perform_node_fail_scenario(self, test_config):
+            failed = str(survived_node.account.ssh_client.exec_command(
+                "grep '%s' %s" % (failed_pattern(), IgniteAwareService.STDOUT_STDERR_CAPTURE))[1].read(),
+                         sys.getdefaultencoding())
+
+            self.logger.warn("Node '%s' (%s) has detected the following failures:%s%s" % (
+                survived_node.name, node_id(survived_node), os.linesep, failed))
+
+            raise AssertionError(
+                "Wrong number of failed nodes: %d. Expected: %d. Check the logs." % (failed_cnt, len(failed_nodes)))
+
+        for service in [srv for srv in self.test_context.services if isinstance(srv, IgniteAwareService)]:
+            for node in [srv_node for srv_node in service.nodes if srv_node not in failed_nodes]:
+                cmd = "grep -i '%s' %s | wc -l" % ("local no1de segmented", IgniteAwareService.STDOUT_STDERR_CAPTURE)
+
+                failed = str(node.account.ssh_client.exec_command(cmd)[1].read(), sys.getdefaultencoding())
+
+                if int(failed) > 0:
+                    raise AssertionError(
+                        "Wrong node failed (segmented) on '%s'. Check the logs." % node.name)
+
+    def setup(self):
+        super().setup()
+
+        self.netfilter_store_path = os.path.join(self.tmp_path_root, "iptables.bak")
+
+        # Store current network filter settings.
+        for node in self.test_context.cluster.nodes:
+            cmd = "sudo iptables-save | tee " + self.netfilter_store_path
+
+            exec_error = str(node.account.ssh_client.exec_command(cmd)[2].read(), sys.getdefaultencoding())
+
+            if "Warning: iptables-legacy tables present" in exec_error:
+                cmd = "sudo iptables-legacy-save | tee " + self.netfilter_store_path
+
+                exec_error = str(node.account.ssh_client.exec_command(cmd)[2].read(), sys.getdefaultencoding())
+
+            assert len(exec_error) == 0, "Failed to store iptables rules on '%s': %s" % (node.name, exec_error)
+
+            self.logger.debug("Netfilter before launch on '%s': %s" % (node.name, dump_netfilter_settings(node)))
+
+    def teardown(self):
+        # Restore previous network filter settings.
+        cmd = "sudo iptables-restore < " + self.netfilter_store_path
+
+        errors = []
+
+        for node in self.test_context.cluster.nodes:
+            exec_error = str(node.account.ssh_client.exec_command(cmd)[2].read(), sys.getdefaultencoding())
+
+            if len(exec_error) > 0:
+                errors.append("Failed to restore iptables rules on '%s': %s" % (node.name, exec_error))
+            else:
+                self.logger.debug("Netfilter after launch on '%s': %s" % (node.name, dump_netfilter_settings(node)))
+
+        if len(errors) > 0:
+            self.logger.error("Failed restoring actions:" + os.linesep + os.linesep.join(errors))
+
+            raise RuntimeError("Unable to restore node states. See the log above.")
+
+        super().teardown()
```

Comment:
should this be at some test_superclass?

Reply:
We have no such superclass for network issues. For now, there are no responsibilities for other test/class to care for netfilter restoring. Also, the test environment clears iptables.

Reply:
AFAIU, @ivandasch need this feature (or it's part) at another test?
If so, let's define the base class.

Reply:
I suggest to split the tickets. Let's made refactoring ticket separate.

Reply:
Ok